File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/SuperHttp.java
package com.ashlikun.okhttputils.http;

import com.ashlikun.okhttputils.http.callback.Callback;

import java.io.IOException;
import java.lang.reflect.Type;

/**
 * 作者　　: 李坤
 * 创建时间:2017/3/17　15:44
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */

public interface SuperHttp {


    //异步回调
    <T> ExecuteCall execute(Callback<T> callback);


    //同步执行
    <ResultType> ResultType syncExecute(Type rawType, Type... typeArguments) throws IOException;


}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/SuperHttp.kt
package com.ashlikun.okhttputils.http

import com.ashlikun.okhttputils.http.HttpException
import com.ashlikun.okhttputils.http.callback.Callback
import java.lang.reflect.Type

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.16 19:25
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：
 */

interface SuperHttp {
    //异步回调
    fun <T> execute(callback: Callback<T>): ExecuteCall

    //同步执行
    @Throws(HttpException::class)
    fun <ResultType> syncExecute(rawType: Type, vararg typeArguments: Type): ResultType
}